protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -188,7 +191,84 @@ impl DaSequencerStorage for Storage {
 	}
 
 	fn produce_next_block(&self) -> Result<Option<SequencerBlock>, DaSequencerError> {
-		todo!()
+		let cf_pending = self.db.cf_handle(cf::PENDING_TRANSACTIONS).ok_or_else(|| {
+			DaSequencerError::StorageAccess("Missing colomn family: pending transactions".into())
+		})?;
+
+		// Collect and deserialize all pending transactions
+		let mut transactions: Vec<Transaction> = Vec::new();
+
+		let iter = self.db.iterator_cf(&cf_pending, rocksdb::IteratorMode::Start);
+		for item in iter {
+			let (_key, value) = item.map_err(|e| DaSequencerError::RocksDbError(e.to_string()))?;
+			let tx: Transaction = bcs::from_bytes(&value)
+				.map_err(|e| DaSequencerError::Deserialization(e.to_string()))?;
+			transactions.push(tx);
+		}
+
+		if transactions.is_empty() {
+			return Ok(None);
+		}
+
+		// Sort transactions using their Ord implementation
+		transactions.sort();
+
+		let mut selected_txs = Vec::new();
+		let mut block_size: u64 = 0;
+
+		for tx in &transactions {
+			let tx_size = bcs::to_bytes(tx)
```

Comment:
You don't have to serialize the Tx because you get them in byte from the DB; Get the size at that time.

Reply:
But, I have to deserialize them in order to sort them, which means I have to serialize again after for building the block.

Reply:
Now that sorting happens lexicographically by using the composite key, I do not need to order unserialized keys, so, it now just deals with serialized values, bytes directly from the DB.